### PR TITLE
porting the vault monkeypatch here

### PR DESCRIPTION
**Pull request by irwaters**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'vault'
+gem 'vault', "0.5.0"
```
Comment (grosser) [inline Gemfile]: `>= 0.5` ?

File: lib/secrets.rb [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+# fixed in vault server 0.6.2 https://github.com/hashicorp/vault/pull/1795
+Vault::Client.prepend(Module.new do
+  def success(response)
+    response.content_type = 'application/json' if response.body && response.body.start_with?('{', '[')
+    super
+  end
+end)
+
+# fixed in vault gem 0.5.1 https://github.com/hashicorp/vault-ruby/pull/93
+if defined?(Vault::Secret)
```
Comment (grosser) [inline lib/secrets.rb]: are we using to_h ?

**Comment by grosser** [comment]
👍

File: lib/secrets.rb (hunk 2 above)
Comment (irwaters) [inline lib/secrets.rb]: just checked, don't see it/removed code.

**Pull request merged**